packages/puppeteer-core/src/node/ScreenRecorder.ts

```diff
@@ -56,25 +62,73 @@ export class ScreenRecorder extends PassThrough {
   #controller = new AbortController();
   #lastFrame: Promise<readonly [Buffer, number]>;
 
+  #fps: number;
+
   /**
    * @internal
    */
   constructor(
     page: Page,
     width: number,
     height: number,
-    {speed, scale, crop, format, path}: ScreenRecorderOptions = {},
+    {
+      speed,
+      scale,
+      crop,
+      format,
+      fps,
+      loop,
+      delay,
+      quality,
+      colors,
+      path,
+    }: ScreenRecorderOptions = {},
   ) {
     super({allowHalfOpen: false});
 
+    format ??= 'webm';
+    fps ??= DEFAULT_FPS;
+    loop ||= -1;
```

Comment:
Can we add comment why we did this, I already forgot when reading the from the start:
Maps 0 to -1 as FFMPEG maps 0 to Infinity.